### ISSUE TITLE
Generating representation vectors for the static documents dataset 

### ISSUE THREAD

**Issue by cn-boop** [facebookresearch/DPR#166]
Hi, I find Generating representation vectors for the static documents dataset  is very slow. How can i fast the speed of this procedure?

**Comment by cn-boop** [comment]
I also try to increase the batch size to 4096, but i find the speed is still low.

**Comment by vlad-karpukhin** [comment]
Yes, increasing the batch size doesn't speed the process after some, pretty low, value.
I think it is because the bottleneck is in preprocessing data step - reading string passages, tokenizing them, convert to tensors, preparing batches, then sending to (presumably working in DP mode) models' forward method where it gets split to multiple processing threads (one per gpu). All these steps are done on cpu and in one thread.
DPR's current approach is to leverage the parallelism of the process - you can just shard it to as many processing nodes as you want, if you have many gpus.
What you can do is to preprocess the wikipedia passage file - tokenize and convert it to tensors - serialize passages as np arrays and then read them as convert to pytorch tensors. This, in theory, should speed up the process.
